### ISSUE TITLE
Compiler warning: Initialization of 'UnsafePointer<Element>' results in a dangling pointer

### ISSUE THREAD

**Issue by josephg** [chinedufn/swift-bridge#108]
The swift compiler emits warnings about this line:

<img width="608" alt="image" src="https://user-images.githubusercontent.com/47413/184812272-11c6f40c-f310-4adf-93be-a952e7f840cd.png">

> SwiftBridgeCore.swift:210:36: Initialization of 'UnsafePointer<Element>' results in a dangling pointer
> SwiftBridgeCore.swift:210:50: Implicit argument conversion from 'Array<Element>' to 'UnsafePointer<Element>' produces a pointer valid only for the duration of the call to 'init(_:)'
> SwiftBridgeCore.swift:210:50: Use the 'withUnsafeBufferPointer' method on Array in order to explicitly convert argument to buffer pointer valid for a defined scope

The warning is totally ignoreable, but it'd be nice to fix.

**Comment by chinedufn** [comment]
Well, that comment that I left is actually wrong. The `GOOD!` example is actually unsound. 

As far as I know Swift does not _guarantee_ that those pointer won't get freed before `useMyPointer` runs.

What we're really supposed to do is get a scoped pointer, like we do for strings. i.e. something like

https://github.com/chinedufn/swift-bridge/blob/b654a24d6d5169e0e73192ebffa556cd43044512/crates/swift-bridge-build/src/generate_core/string.swift#L79-L90

---

So, long story short this `toUnsafeBufferPointer` method is incorrect and shouldn't exist.

For now we can actually just delete this since it's not used anywhere since we don't yet support bridging a Swift Array to Rust.

If you're up for it, feel free to fix this warning by deleting the function entirely. 
This function is unsound so I won't publish it as a breaking change, especially since I doubt that anyone is actually using it.
